blob/fileblob/fileblob.go

```diff
@@ -565,6 +582,20 @@ func (b *bucket) Delete(ctx context.Context, key string) error {
 }
 
 func (b *bucket) SignedURL(ctx context.Context, key string, opts *driver.SignedURLOptions) (string, error) {
-	// TODO(Issue #546): Implemented SignedURL for fileblob.
+	path, err := b.path(key)
+	if err != nil {
+		return "", err
+	}
+
+	return b.urlSigner.SignedURL(ctx, key, path, opts)
+}
+
+type URLSigner interface {
+	SignedURL(ctx context.Context, key string, path string, opts *driver.SignedURLOptions) (string, error)
+}
+
+type errorURLSigner struct{}
```

Comment:
And while this is unexported, it would be good to include a brief description of this type. Perhaps:

```suggestion
// errorURLSigner is a stub implementation of URLSigner used when no URLSigner is passed to OpenBucket.
type errorURLSigner struct{}
```